app/models/category.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Category < ApplicationRecord
+  validates :title, presence: true
+
+  has_many :category_rewards, dependent: :restrict_with_error
```

Comment:
The same for the categories. If the rewards have a category or vice versa, they cannot be deleted.